src/libraries/Common/src/System/Security/Cryptography/MLDsaImplementation.Windows.cs

```diff
@@ -2,47 +2,216 @@
+                CryptoPool.Return(keyBlob, clearSize: 0);
+            }
+        }
+
+        protected override void ExportMLDsaSecretKeyCore(Span<byte> destination)
+        {
+            ArraySegment<byte> keyBlob = Interop.BCrypt.BCryptExportKey(_key, Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_BLOB);
+
+            try
+            {
+                ReadOnlySpan<byte> keyBytes = PqcBlobHelpers.DecodeMLDsaBlob(
+                    keyBlob,
+                    out ReadOnlySpan<char> parameterSet,
+                    out string blobType);
+
+                Debug.Assert(blobType == Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_BLOB);
 
-        protected override void ExportMLDsaPublicKeyCore(Span<byte> destination) =>
-            throw new PlatformNotSupportedException();
+                // Length is known, but we'll slice just in case
+                MLDsaAlgorithm algorithm = PqcBlobHelpers.GetMLDsaAlgorithmFromParameterSet(parameterSet);
+                Debug.Assert(keyBytes.Length == algorithm.SecretKeySizeInBytes);
+                keyBytes.Slice(0, algorithm.SecretKeySizeInBytes).CopyTo(destination);
+            }
+            finally
+            {
+                CryptoPool.Return(keyBlob, clearSize: keyBlob.Count);
+            }
+        }
 
-        protected override void ExportMLDsaSecretKeyCore(Span<byte> destination) =>
-            throw new PlatformNotSupportedException();
+        protected override void ExportMLDsaPrivateSeedCore(Span<byte> destination)
+        {
+            ArraySegment<byte> keyBlob = Interop.BCrypt.BCryptExportKey(_key, Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_SEED_BLOB);
 
-        protected override void ExportMLDsaPrivateSeedCore(Span<byte> destination) =>
-            throw new PlatformNotSupportedException();
+            try
+            {
+                ReadOnlySpan<byte> keyBytes = PqcBlobHelpers.DecodeMLDsaBlob(
+                    keyBlob,
+                    out ReadOnlySpan<char> parameterSet,
+                    out string blobType);
 
-        protected override bool TryExportPkcs8PrivateKeyCore(Span<byte> destination, out int bytesWritten) =>
-            throw new PlatformNotSupportedException();
+                Debug.Assert(blobType == Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_SEED_BLOB);
 
-        internal static partial MLDsaImplementation GenerateKeyImpl(MLDsaAlgorithm algorithm) =>
-            throw new PlatformNotSupportedException();
+                // Length is known, but we'll slice just in case
+                MLDsaAlgorithm algorithm = PqcBlobHelpers.GetMLDsaAlgorithmFromParameterSet(parameterSet);
+                Debug.Assert(keyBytes.Length == algorithm.PrivateSeedSizeInBytes);
+                keyBytes.Slice(0, algorithm.PrivateSeedSizeInBytes).CopyTo(destination);
+            }
+            finally
+            {
+                CryptoPool.Return(keyBlob, clearSize: keyBlob.Count);
```

Comment:
The default is count.

```suggestion
                CryptoPool.Return(keyBlob);
```